crafting-dead-immerse/src/main/java/com/craftingdead/immerse/world/level/block/entity/BaseCenterBlockEntity.java

```diff
@@ -98,18 +109,66 @@ public void removed() {
     }
   }
 
+  public void playerPlacedBlock(LivingExtension<?, ?> player, BlockPos... posArray) {
+    for (BlockPos pos : posArray) {
+      this.playerBlocks.add(pos.immutable());
+    }
+  }
+
+  public void playerRemovedBlock(LivingExtension<?, ?> player, BlockPos... posArray) {
+    for (BlockPos pos : posArray) {
+      this.playerBlocks.remove(pos);
+    }
+  }
+
+  public void destroyPlayerBlocks() {
+    var bedrock = Blocks.BEDROCK.defaultBlockState();
+    var air = Blocks.AIR.defaultBlockState();
+
+    // The UPDATE_KNOWN_SHAPE prevent block neighbors from being updated
+    // we need it to avoid certain blocks from dropping
+    for (BlockPos pos : playerBlocks) {
```

Comment:
This should probably be split across multiple ticks to avoid delaying the game loop for too long (see `BaseLandManager::removeLandClaim`).